pkg/ingester/user_state.go

```diff
@@ -50,82 +49,62 @@ func (cfg *UserStatesConfig) RegisterFlags(f *flag.FlagSet) {
 
 func newUserStates(cfg *UserStatesConfig) *userStates {
 	return &userStates{
-		states: map[string]*userState{},
-		cfg:    cfg,
+		cfg: cfg,
 	}
 }
 
 func (us *userStates) cp() map[string]*userState {
-	us.mtx.RLock()
-	defer us.mtx.RUnlock()
-	states := make(map[string]*userState, len(us.states))
-	for id, state := range us.states {
-		states[id] = state
-	}
+	states := make(map[string]*userState, us.numUsers())
```

Comment:
This will range over the map twice - might be more costly than not pre-declaring the size of the map?

Reply:
I agree, I'll change it.  It looks like `cp()` can be eliminated now we don't ever lock the map, but that can be a separate change.